### PR TITLE
Delete items from shopping list

### LINKED ISSUE #11: 11. As a user, I want to be able to delete items from my shopping list so that my list isn’t cluttered with items I don’t want to buy in the future.
It’s possible our users may make an error when entering an item or may decide they won’t be buying a certain item again in the future. Let’s help them keep their lists tidy!

AC:

* User is able to delete an item from the shopping list
* Before deleting, prompt the user to confirm that they really want to delete the item to prevent accidental deletions
* Deletion should cause the associated record(s) in the database to be deleted

Notes:

* As noted in the wireframe https://github.com/the-collab-lab/smart-shopping-list/blob/main/_resources/wireframes/list-populated.png activating the "Delete" link should pop-up a confirmation dialog because this is a "destructive" action.
* Many teams jump to a modal as the solution here, but modals are fraught with accessibility concerns, so consider using a simple JavaScript confirm dialog.

### PR DESCRIPTION
**Pull request by sparsell**
## Description

- Added a "Delete" button to each `listItem` on List.jsx 
- When clicked, a JS pop-up dialogue box asks if the user is sure they want to delete?
- Used firebase `deleteDoc` function, which gets called when user clicks on "Delete" button

## Related Issue

Closes #11 

## Acceptance Criteria

- User is able to delete an item from the shopping list
- Before deleting, prompt the user to confirm that they really want to delete the item to prevent accidental deletions
- Deletion should cause the associated record(s) in the database to be deleted

## Type of Changes

|     | Type                       |
| --- | -------------------------- |
|    | :bug: Bug fix              |
|  ✓ | :sparkles: New feature     |
|    | :hammer: Refactoring       |
|    | :100: Add tests            |
|    | :link: Update dependencies |
|    | :scroll: Docs              |

### Before
Screen Shot 2022-05-12 at 9 58 29 AM

### After
Screen Shot 2022-05-12 at 9 55 07 AM

- added styling to list by removing bullet point and using flexbox to align checkboxes and "Delete" buttons

## Testing Steps / QA Criteria
- checkout branch `sp-ks-deleteItem`
- create a new list and add some items if you don't already have a list showing 
- click the "Delete" button for an item and note the dialogue box message:
 -- choose "Cancel" the first time and you should return to the list with no change
 -- choose "OK" the second time and you should not see the item in the list anymore
- check the item has also been deleted from Firestore

### CODE REVIEW

**Review by artsycoder533 — APPROVED** [review]
I already had a list saved.  I removed a list item and an alert was soon asking if I was sure if I wanted to delete the item.  When I click ok, the item is removed from the list in the UI as well as in firebase.  When I click delete on another item and hit cancel the alert goes away and the UI is unchanged as well as in firebase.

**Review by daniel-sj-hong — APPROVED** [review]
Items are being deleted as intended! Nice work!

File: src/views/List/List.jsx [hunk 1]
```diff
@@ -117,6 +123,18 @@ export default function List({ token }) {
     setSearchInput(value);
   };
 
+  //onClick handler for delete button
+  async function deleteItem(listItem) {
+    const docRef = doc(db, token, listItem.id);
+    console.log(docRef);
+    const confirm = window.confirm(
+      `Do you want to delete ${listItem.name} from your list?`,
```
Comment (lindseyindev) [inline src/views/List/List.jsx:137]: Nice delete function, great inclusion of the specific item name in the message to the user 🌞

very logical to step through with the confirm variable and then checking if the user returns true

File: src/views/List/List.css [hunk 2]
```diff
@@ -23,3 +23,26 @@
 .message {
   margin-top: 2rem;
 }
+
+.list-item {
```
Comment (lindseyindev) [inline src/views/List/List.css:27]: great work on adding styling after finishing the delete functionality, you had extra time and it was used really well, the app looks really nice and clean and you're setting up the foundation to make further customizations easier

**Review by lindseyindev — APPROVED** [review]
Works perfectly and the styling is great, your team finished with the delete functionality early and had extra time and you added in a lot of styling details and foundation to the list. 

As we've seen it's not always possible to finish early due to the size and complexity of issues but going above the acceptance criteria when having additional time and making changes that will make your team's life easier and help you complete your next issues faster (styling and UI) is a huge success and I hope you two own that (especially in any future interviews you have 😼)

With my approval you are good to merge!!

**Pull request merged**